nft_ingester/src/tasks/common/mod.rs

```diff
@@ -110,24 +111,29 @@ impl BgTask for DownloadMetadataTask {
             id: Unchanged(download_metadata.asset_data_id.clone()),
             metadata: Set(body),
             reindex: Set(Some(false)),
+            download_metadata_seq: Set(Some(download_metadata.seq)),
```

Comment:
I am not sure about this addition. I can see that the download_metadata uri should be protected by a seq since that is on chain. But the offchain metadata can change at any time. To me the system should always download the latest metadata that is offchain irrespectively of the seq?

So as long as we know that the URI in the asset_data table is only ever updated to the latest we shouldn't protect the download metadata itself? 

If the URI is updated, then indeed we must redownload metadata but I also don't see a harm in scheduling additional metadata downloads in between. Not having an additional sequence number to check might be quite good also.

Adding this sequence number has limited value and causes extra DB queries/complexity.

Reply:
If we want to limit the number of downloads, we could either

a) keep track of URI changes and only trigger a redownload if the URI hasn't been downloaded yet
b) make the download metadata task smarter to understand if the offchain data has changed (i.e. do some kind of HEAD check to find the HTTP headers to show if data is changed)

Reply:
>I can see that the download_metadata uri should be protected by a seq since that is on chain. But the offchain metadata can change at any time. To me the system should always download the latest metadata that is offchain irrespectively of the seq?

So the URI is already protected by its own sequence number in the `mint_V1` and `update_metadata` code, but the reason I added this seq is to make sure that a task doesn't update `asset_data.metadata` with downloaded data from an older task.

I think because URI can be changed, this can happen whether stuff is indexed in-order or out-of-order.

Example:
1. `mint_v1`, has uri1, seq 1, kicks off task(uri1, seq=1).
2. `update_metadata`, has uri2, seq2, kicks of task(uri2, seq=2)
3. Task2 finishes and asset_data.metadata to the data from uri2.
4. Task1 finishes and **incorrectly** sets `asset_data.metadata` to the data from uri1.

However, with using a download_metadata_seq, step 4 would be different.  Task1 would finish and NOT update `asset_data.metadata` to the data from uri1.